File: php/validate.php
<?php
// Validate a Factur-X PDF using the InvoiceXML API.
// Get API key: https://www.invoicexml.com/account/authentication
// Docs:        https://www.invoicexml.com/docs/api/validate/facturx

$apiKey = 'YOUR_API_KEY';

$ch = curl_init('https://api.invoicexml.com/v1/validate/facturx');
curl_setopt_array($ch, [
    CURLOPT_RETURNTRANSFER => true,
    CURLOPT_POST           => true,
    CURLOPT_POSTFIELDS     => [
        'file'    => new CURLFile('invoice.pdf', 'application/pdf'),
        'version' => '2.3.2',
        'profile' => 'extended',
    ],
    CURLOPT_HTTPHEADER     => ['Authorization: Bearer ' . $apiKey],
]);

$report = curl_exec($ch);
$status = curl_getinfo($ch, CURLINFO_HTTP_CODE);
curl_close($ch);

if ($status !== 200) {
    fwrite(STDERR, "InvoiceXML API error $status: $report\n");
    exit(1);
}

echo $report;

File: java/Validate.java
// Validate a Factur-X PDF using the InvoiceXML API.
// Get API key: https://www.invoicexml.com/account/authentication
// Docs:        https://www.invoicexml.com/docs/api/validate/facturx
//
// Dependency: com.squareup.okhttp3:okhttp:4.12.0

import okhttp3.*;
import java.io.File;

public class Validate {
    public static void main(String[] args) throws Exception {
        String apiKey = "YOUR_API_KEY";

        RequestBody body = new MultipartBody.Builder()
            .setType(MultipartBody.FORM)
            .addFormDataPart("file", "invoice.pdf",
                RequestBody.create(new File("invoice.pdf"), MediaType.parse("application/pdf")))
            .addFormDataPart("version", "2.3.2")
            .addFormDataPart("profile", "extended")
            .build();

        Request request = new Request.Builder()
            .url("https://api.invoicexml.com/v1/validate/facturx")
            .header("Authorization", "Bearer " + apiKey)
            .post(body)
            .build();

        try (Response response = new OkHttpClient().newCall(request).execute()) {
            String report = response.body().string();
            if (!response.isSuccessful()) {
                System.err.println("InvoiceXML API error " + response.code() + ": " + report);
                System.exit(1);
            }
            System.out.println(report);
        }
    }
}
